Instrumenting forks of create-react-app: cannot find module webpack.config.js

Thanks for this awesome project!

I'd like to instrument a project which uses a fork of `create-react-app` with only the slightest change: it [adds 2 lines in `webpack.config.js`](https://github.com/bitauth/create-react-app-bitauth-ide/commit/0bef6a377308e5149ee3a25f3d0bbdd15f6f664a), and is otherwise unmodified. (See [`Alternatives to Ejecting`](https://create-react-app.dev/docs/alternatives-to-ejecting/) in the Create React App docs.)

Here's what our [`package.json` looks like](https://github.com/bitauth/bitauth-ide/blob/master/package.json#L6) – we've followed the advice from the Create React App documentation: instead of ejecting, we've replaced `"react-scripts": "3.4.1"` with `"@bitauth/react-scripts-bitauth-ide": "^3.4.1"`.

Could `@cypress/instrument-cra` somehow support instrumenting `react-scripts` with a different `webpackConfigPath`? 

Currently, when I set the `start` script to `react-scripts -r @cypress/instrument-cra start`, the `start` task returns:

```
Error: Cannot find module '[...]/node_modules/react-scripts/config/webpack.config.js'
```

But when I modify `node_modules/@cypress/instrument-cra/index.js` from:
```js
const webpackConfigPath = path.resolve(
  findYarnWorkspaceRoot() || process.cwd(),
  'node_modules',
  'react-scripts',
  'config',
  'webpack.config.js'
)
```
to:
```js
const webpackConfigPath = path.resolve(
  findYarnWorkspaceRoot() || process.cwd(),
  'node_modules',
  '@bitauth',
  'react-scripts-bitauth-ide',
  'config',
  'webpack.config.js'
)
```
Instrumentation is successful, and everything works perfectly.

Would you consider accepting this path as a configuration parameter somewhere?

Maybe `webpackConfigPath` in `cypress.json` or `cypressWebpackConfigPath` in `package.json`?

--- a/README.md
+++ b/README.md
@@ -37,6 +37,16 @@ If you want to exclude files from coverage, for example `src/serviceWorker.js`, 
 }
 ```
 
+## Instrument a fork
+
+To instrument a [fork of `react-scripts`](https://create-react-app.dev/docs/alternatives-to-ejecting/), provide the path to the new `webpack.config.js` in your `package.json` as `cypressWebpackConfigPath`, e.g.:
+
+```json
+{
+  "cypressWebpackConfigPath": "./node_modules/@my-org/my-react-scripts-fork/config/webpack.config.js"
+}
+```
+
 ## Debugging
 
 Run with environment variable `DEBUG=instrument-cra` to see the verbose logs

--- a/index.js
+++ b/index.js
@@ -2,13 +2,29 @@ const debug = require('debug')('instrument-cra')
 const path = require('path')
 const findYarnWorkspaceRoot = require('find-yarn-workspace-root')
 
-const webpackConfigPath = path.resolve(
-  findYarnWorkspaceRoot() || process.cwd(),
-  'node_modules',
-  'react-scripts',
-  'config',
-  'webpack.config.js'
-)
+const workspaceRoot = findYarnWorkspaceRoot() || process.cwd()
+const packagePath = path.resolve(workspaceRoot, 'package.json')
+
+let package = { cypressWebpackConfigPath: undefined }
+try {
+  package = require(packagePath)
+} catch {
+  debug('failed to read package.json at path: %s', packagePath)
+}
+
+const webpackConfigPath =
+  package.cypressWebpackConfigPath !== undefined
+    ? path.resolve(
+        workspaceRoot,
+        path.normalize(package.cypressWebpackConfigPath)
+      )
+    : path.resolve(
+        workspaceRoot,
+        'node_modules',
+        'react-scripts',
+        'config',
+        'webpack.config.js'
+      )
 
 debug('path to react-scripts own webpack.config.js: %s', webpackConfigPath)
 

